ScriptureRenderingPipeline/Renderers/ScripturalMarkdownRendererBase.cs

```diff
@@ -14,196 +14,219 @@
-                    }
-
-                    Directory.CreateDirectory(Path.Join(destinationDir, book.BookId));
-                    outputTasks.Add(File.WriteAllTextAsync(Path.Join(destinationDir, book.BookId, $"{chapter.ChapterNumber}.html"), builder.ToString()));
-                    printBuilder.Append(builder);
-                    outputBook.Chapters.Add(new OutputChapters()
-                    {
-                        Label = chapter.ChapterNumber,
-                        Number = chapter.ChapterNumber,
-                    });
-                }
-                outputIndex.Bible.Add(outputBook);
+              headingBlock.Level++;
             }
-            outputTasks.Add(File.WriteAllTextAsync(Path.Join(destinationDir, "index.json"), JsonSerializer.Serialize(outputIndex)));
 
-            if (books.Count > 0)
+            foreach (var link in parsedVerse.Descendants<LinkInline>())
             {
-                outputTasks.Add(File.WriteAllTextAsync(Path.Join(destinationDir, "print_all.html"), printTemplate.Render(Hash.FromAnonymousObject(new { content = printBuilder.ToString(), heading }))));
+              if (link.Url == null)
+              {
+                continue;
+              }
+              if (link.Url != null && link.Url.EndsWith(".md"))
+              {
+                link.Url = RewriteContentLinks(link.Url, tnBook, tnChapter);
+              }
             }
 
-            await Task.WhenAll(outputTasks);
+            var tnVerse = new TranslationMaterialsVerse(Path.GetFileNameWithoutExtension(file), parsedVerse.ToHtml(pipeline));
+            tnChapter.Verses.Add(tnVerse);
+          }
+          tnBook.Chapters.Add(tnChapter);
+        }
+        output.Add(tnBook);
+      }
+      return output;
+    }
+    public virtual async Task RenderAsync(ZipFileSystem sourceDir, string basePath, string destinationDir,
+         Template printTemplate, string repoUrl, string heading, string baseUrl,
+        string userToRouteResourcesTo, string textDirection, string languageCode, string languageName, bool isBTTWriterProject = false)
+    {
+      var books = await LoadMarkDownFilesAsync(sourceDir, basePath, baseUrl, userToRouteResourcesTo, languageCode);
+      var printBuilder = new StringBuilder();
+      var outputTasks = new List<Task>();
+      var outputIndex = new OutputIndex()
+      {
+        LanguageCode = languageCode,
+        TextDirection = textDirection,
+        RepoUrl = repoUrl,
+        LanguageName = languageName,
+        ResourceType = ContentType,
+        ResourceTitle = heading,
+        Bible = new List<OutputBook>(),
+      };
+      var downloadIndex = new DownloadIndex()
+      {
+        Data = new List<OutputBook>()
+      };
+      foreach (var book in books)
+      {
+        var outputBook = new OutputBook()
+        {
+          Label = book.BookName,
+          Slug = book.BookId
+        };
+        var outPutBookDataIncluded = new OutputBook()

```

Comment:
typo: should be "outputBookDataIncluded".

Also, I'd prefer a different term than "Data" -- what is this data?  Rendered content?  Maybe it should be "outputBookContentIncluded"?